File: 2022/10_cathode_ray_tube/part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func printReg(d int, X int) {
	cycles := d % 40
	if X == cycles || X+1 == cycles || X+2 == cycles {
		fmt.Print("#")
	} else {
		fmt.Print(".")
	}
	if cycles == 0 {
		fmt.Println()
	}
}

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer input.Close()

	X := 1
	cycles := 1

	scanner := bufio.NewScanner(input)
	for scanner.Scan() {
		line := strings.Fields(scanner.Text())
		inst := line[0]
		if inst == "addx" {
			val, _ := strconv.Atoi(line[1])
			printReg(cycles, X)
			cycles++
			printReg(cycles, X)
			cycles++
			X += val
		} else {
			printReg(cycles, X)
			cycles += 1
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Println(err)
	}
}

File: 2022/10_cathode_ray_tube/part2.py
def printReg(d, X):
    cycles = d % 40
    if X == cycles or X + 1 == cycles or X + 2 == cycles:
        print("#", end="")
    else:
        print(".", end="")
    if cycles == 0:
        print()

with open("input.txt", "r") as file:
    X = 1
    cycles = 1

    for line in file:
        line = line.strip()
        line = line.split()
        inst = line[0]
        if inst == "addx":
            val = int(line[1])
            printReg(cycles, X)
            cycles += 1
            printReg(cycles, X)
            cycles += 1
            X += val
        else:
            printReg(cycles, X)
            cycles += 1